base_classes/NXcg_primitive.nxdl.xml

```diff
@@ -0,0 +1,229 @@
+        <doc>
+             True if the center is a center of mass.
+        </doc>
+        <dimensions rank="1">
+            <dim index="1" value="c"/>
+        </dimensions>
+    </field>
+    <field name="shape" type="NX_NUMBER" units="NX_LENGTH">
+        <doc>
+             A qualitative description of the shape of each primitive.
+        </doc>
+        <dimensions rank="2">
+            <dim index="1" value="c"/>
+            <dim index="2" value="d"/>
+        </dimensions>
+    </field>
+    <field name="length" type="NX_NUMBER" units="NX_LENGTH">
+        <doc>
+             Qualifier for the length of characteristic features of the primitive.
+             
+             Often the term length is associated with the assumption that one
+             edge is parallel to an axis of the coordinate system.
+        </doc>
+        <dimensions rank="1">
+            <dim index="1" value="c"/>
+        </dimensions>
+    </field>
+    <field name="width" type="NX_NUMBER" units="NX_LENGTH">
+        <doc>
+             Qualifier often used to describe the length of one characteristic edge
+             within the coordinate system.
+        </doc>
+        <dimensions rank="1">
+            <dim index="1" value="c"/>
+        </dimensions>
+    </field>
+    <field name="is_closed" type="NX_BOOLEAN">
+        <doc>
+             True if primitive is closed such that it has properties like area or volume.
+        </doc>
+        <dimensions rank="1">
+            <dim index="1" value="c"/>
+        </dimensions>
+    </field>
+    <field name="volume" type="NX_NUMBER" units="NX_VOLUME">
+        <doc>
+             Volume of each primitive.
+             
+             Set to NaN if does not apply for primitives for which is_closed is False.
+        </doc>
+        <dimensions rank="1">
+            <dim index="1" value="c"/>
+        </dimensions>
+    </field>
+    <field name="area" type="NX_NUMBER" units="NX_AREA">
+        <doc>
+             Alias for surface_area of each primitive.
+             
+             Set to NaN if does not apply for primitives for which is_closed is False.
+        </doc>
+        <dimensions rank="1">
+            <dim index="1" value="c"/>
+        </dimensions>
+    </field>
+    <field name="orientation" type="NX_NUMBER" units="NX_DIMENSIONLESS">
+        <doc>
+             Direction unit vector which points along the
+             longest principal axis of each primitive.
+             
+             Use the depends_on attribute to specify in which coordinate system
+             these direction unit vectors are defined.
+        </doc>
+        <dimensions rank="2">
+            <dim index="1" value="c"/>
+            <dim index="2" value="d"/>
+        </dimensions>
+    </field>
+    <field name="is_mesh" type="NX_BOOLEAN">
+      <doc>
+        Do the primitives define a mesh.
+      </doc>
+    </field>
+    <!--<field name="is_triangle_mesh" type="NX_BOOLEAN">
```

Comment:
Remove commented out fields

Reply:
Nahh, maybe we should propose them as well but with is_surface_mesh being more useful and general than is_triangle_mesh